app/src/main/java/comp3350/mealbuddy/objects/consumables/FoodIntPair.java

```diff
@@ -1,11 +1,20 @@
 package comp3350.mealbuddy.objects.consumables;
 
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+
+import java.util.Objects;
+
 public class FoodIntPair {
 
     public Food food;
     public Integer quantity;
 
     public FoodIntPair(Food food, Integer quantity) {
+        if (quantity < 0) {
+            quantity = 0;
```

Comment:
this doesn't make sense for a FoodIntPair, since in a meal there should never be "0" of a food. Make this throw an exception.

Reply:
Alright

Reply:
No it does make sense.  Since if we set the quantity to zero it should remove the food from the meal

Reply:
We just shouldn't add the food then, since a quantity of 0 means the food doesn't exist.

Reply:
If they pass a bad value, there should be an exception for that since the user passed a bad value. It's okay to have errors like this since we are deliberately saying "you must pass a valid value"

Reply:
I think I agree with you Arvin,if the quantity is <=0 , the food shouldn't be added in the first place

Reply:
Aight.  Just make sure you rerun the tests and modfy any that fail to expect this exception behaviour

Reply:
@prettyboyamin You will have to make this change, run all the tests, make sure they all pass, and then re push.  Then you can resolve this conversation